### PR TITLE
Trigger contrib build

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/35481

## Description
Added support to triggering a  contribution build in gitlab using the webhook integration in Content Gold. 

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 1]
```diff
@@ -680,6 +726,9 @@ def main():
     url = params.get('url')
     verify_certificate = not params.get('insecure', False)
     proxy = params.get('proxy', False)
+    # you can find this trigger in https://vault.paloaltonetworks.local/home#R2VuZXJpY1NlY3JldERldGFpbH
+    # M6RGF0YVZhdWx0Ojc5NjU1N2RmLTg5YzQtNGQ5Yi04YmNlLTljNTMwZWU1NjVjYTpSZWNvcmRJbmRleDo1OklzVHJ1bmNhdGVk
```
Comment (yaakovi) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:730]: remove

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 2]
```diff
@@ -710,7 +759,8 @@ def main():
             'gitlab-issue-create': gitlab_create_issue_command,
             'gitlab-issue-edit': gitlab_edit_issue_command,
             'gitlab-group-projects-list': gitlab_group_projects_list_command,
-            'gitlab-raw-file-get': gitlab_get_raw_file_command
+            'gitlab-raw-file-get': gitlab_get_raw_file_command,
+            'gitlab-trigger-contribution-build': gitlab_trigger_contribution_build_command
```
Comment (yaakovi) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:763]:
```suggestion
            'gitlab-trigger-contribution-build': gitlab_trigger_contribution_build_command,
```

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 3]
```diff
@@ -184,6 +185,18 @@ def get_raw_file_request(self, project_id: str, file_path: str, ref: str):
         response = response.strip("'").strip('"')
         return response
 
+    def trigger_contrib_build(self, base_branch: str, contrib_branch: str, pr_number: int, token: str):
```
Comment (yaakovi) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:188]: no need to pass the token here, it is already part of the Client object

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 4]
```diff
@@ -184,6 +185,18 @@ def get_raw_file_request(self, project_id: str, file_path: str, ref: str):
...
+        headers = self._headers
+        data = {
+            'token': token,
+            'ref': base_branch,
+            'variables[CONTRIB_BRANCH]': contrib_branch,
+            'variables[PULL_REQUEST_NUMBER]': pr_number,
+            'variables[CI_COMMIT_BRANCH]': base_branch,
+            'variables[CI_PIPELINE_SOURCE]': 'contrib'
+        }
+        return self._http_request('post', url_suffix='projects/2596/trigger/pipeline', data=data, headers=headers)
```
Comment (yaakovi) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:198]: the project ID should be an argument

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 5]
```diff
@@ -184,6 +185,18 @@ def get_raw_file_request(self, project_id: str, file_path: str, ref: str):
...
 
+    def trigger_contrib_build(self, base_branch: str, contrib_branch: str, pr_number: int, token: str):
+        headers = self._headers
+        data = {
+            'token': token,
+            'ref': base_branch,
+            'variables[CONTRIB_BRANCH]': contrib_branch,
+            'variables[PULL_REQUEST_NUMBER]': pr_number,
+            'variables[CI_COMMIT_BRANCH]': base_branch,
+            'variables[CI_PIPELINE_SOURCE]': 'contrib'
```
Comment (yaakovi) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:196]: both argument names and values should be configurable by the integration arguments

**Review by yaakovi — CHANGES_REQUESTED** [review]
only looked at the integration changes

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 6]
```diff
@@ -1058,7 +1062,78 @@ script:
...
+      isArray: false
+      name: trigger_variables
+    - default: false
+      description: Project ID.
+      isArray: false
+      name: project_id
+      required: true
+      secret: false
+    description: Trigger a contribution build
+    name: gitlab-trigger-build
```
Comment (yaakovi) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:1083]:
```suggestion
    name: gitlab-trigger-pipeline
```

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 7]
```diff
@@ -1058,7 +1062,78 @@ script:
...
+    - contextPath: GitLab.Pipeline.user.state
+      description: State of the user who triggered the pipeline.
+      type: String
+    - contextPath: GitLab.Pipeline.user.avatar_url
+      description: Avatar URL of the user who trigerred the pipeline.
+      type: String
+    - contextPath: GitLab.Pipeline.user.web_url
+      description: Web URL of the user who triggered the pipeline.
+      type: String
+  dockerimage: demisto/python3:3.10.1.25933
```
Comment (yaakovi) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:1136]: revert to 3.9

File: Packs/DevSecOps/ReleaseNotes/1_0_7.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### GitLab
+- Added the command **gitlab-trigger-contribution-build** to trigger a contribution build in Gitlab.
```
Comment (yaakovi) [inline Packs/DevSecOps/ReleaseNotes/1_0_7.md:4]:
```suggestion
- Added the ***gitlab-trigger-pipeline*** command to trigger a pipeline in Gitlab.
```

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 9]
```diff
@@ -680,6 +726,7 @@ def main():
     url = params.get('url')
     verify_certificate = not params.get('insecure', False)
     proxy = params.get('proxy', False)
+    trigger_token = params.get('trigger_token', '')
```
[the remark this answers is not in the document]
Comment (merit-maita) [reply Packs/DevSecOps/Integrations/GitLab/GitLab.py:729]: the api_token is the private api key generated by user, the trigger token, is one specific token for triggering builds in giltab.
they have 2 different values.

File: Packs/DevSecOps/ReleaseNotes/1_0_7.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### GitLab
+- Added the ***gitlab-trigger-pipeline*** command to trigger a pipeline in Gitlab.
+- Updated the Docker image to: *demisto/python3:3.9.9.25564*.
```
Comment (yaakovi) [inline Packs/DevSecOps/ReleaseNotes/1_0_7.md:5]: update to actual image

**Review by yaakovi — APPROVED** [review]
please schedule a 15 minutes demo

File: Packs/GitHub/Integrations/GitHub/GitHub.py [hunk 11]
```diff
@@ -1937,6 +1937,35 @@ def github_releases_list_command():
...
+def github_update_comment_command():
+    args = demisto.args()
+    comment_id = args.get('comment_id')
+    issue_number = args.get('issue_number')
+    body = args.get('body')
+    response = update_comment(comment_id, body)
+
+    ec_object = format_comment_outputs(response, issue_number)
+    ec = {
+        'GitHub.Comment(val.IssueNumber === obj.IssueNumber && val.ID === obj.ID)': ec_object
```
Comment (yaakovi) [inline Packs/GitHub/Integrations/GitHub/GitHub.py:1955]:
```suggestion
        'GitHub.Comment(val.IssueNumber === obj.IssueNumber && val.ID === obj.ID)': ec_object,
```

File: Packs/GitHub/Integrations/GitHub/GitHub.py [hunk 12]
```diff
@@ -1937,6 +1937,35 @@ def github_releases_list_command():
...
+    }
+    human_readable = tableToMarkdown('Updated Comment', ec_object, removeNull=True)
+    return_outputs(readable_output=human_readable, outputs=ec, raw_response=response)
+
+
+def github_delete_comment_command():
+    args = demisto.args()
+    comment_id = args.get('comment_id')
+    suffix = ISSUE_SUFFIX + f'/comments/{comment_id}'
```
Comment (yaakovi) [inline Packs/GitHub/Integrations/GitHub/GitHub.py:1964]:
```suggestion
    suffix = f'{ISSUE_SUFFIX}/comments/{comment_id}'
```

File: Packs/GitHub/Integrations/GitHub/GitHub.py [hunk 13]
```diff
@@ -1937,6 +1937,35 @@ def github_releases_list_command():
     return_results(result)
 
 
+def update_comment(comment_id: Union[int, str], msg: str) -> dict:
+    suffix = ISSUE_SUFFIX + f'/comments/{comment_id}'
```
Comment (yaakovi) [inline Packs/GitHub/Integrations/GitHub/GitHub.py:1941]:
```suggestion
    suffix = f'{ISSUE_SUFFIX}/comments/{comment_id}'
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**